src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -439,7 +445,7 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
     Outputs="$(MonoAndroidAssetsDirIntermediate)xa-internal/xa-mam-mapping.xml">
   <MakeDir
       Condition=" '@(_AndroidMamMappingFile->Count())' != '0' "
-      Directories="$(MonoAndroidAssetsDirIntermediate)/xa-internal"
+      Directories="$(MonoAndroidAssetsDirIntermediate)xa-internal"
   />
```

Comment:
Wait, does this really fix it? MSBuild does things in passes, a simplification/example of the ordering:

1. Imports & Properties
2. Item Groups
3. Targets

Would `$(MonoAndroidAssetsPrefix)` have always happened in pass 1, and these targets are pass 3?

Reply:
While `$(MonoAndroidAssetsDirIntermediate)` would always be set in pass 1, there is still an ordering issue *between properties*, because properties are set in file order:

 1. `$(_XARemapMembersFilePath)` was set *first*: https://github.com/xamarin/xamarin-android/blob/69b4ab056badcb5d25a5faf39d08baeb1b8b7010/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L458
 2. `$(MonoAndroidAssetsDirIntermediate)` was set afterward: https://github.com/xamarin/xamarin-android/blob/69b4ab056badcb5d25a5faf39d08baeb1b8b7010/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L885

When (1) is evaluated, `$(MonoAndroidAssetsDirIntermediate)` hasn't been set yet, thus `$(_XARemapMembersFilePath)` will be `xa-internal/xa-remap-members.xml` (which has the nice property of matching the bug report).

Indeed, when viewing the diagnostic build log, we see:

```
_XARemapMembersFilePath = xa-internal/xa-remap-members.xml
```

By moving `$(MonoAndroidAssetsDirIntermediate)` to be defined *before* `$(_XARemapMembersFilePath)` is defined, we ensure that `$(MonoAndroidAssetsDirIntermediate)` doesn't expand to the empty string, because it will have been set by the time it is used.